sdk-base/src/main/java/com/mapbox/maps/plugin/animation/MapAnimationOptions.kt

```diff
@@ -71,49 +85,44 @@ class MapAnimationOptions private constructor(
     /**
      * Set the animator start / cancel / end listener.
      */
-    fun animatorListener(animatorListener: Animator.AnimatorListener) = apply { this.animatorListener = animatorListener }
+    fun animatorListener(animatorListener: Animator.AnimatorListener) =
+      apply { this.animatorListener = animatorListener }
 
     /**
      * Build an actual [MapAnimationOptions] object.
      */
-    fun build(): MapAnimationOptions = MapAnimationOptions(owner, duration, interpolator, animatorListener)
+    fun build(): MapAnimationOptions =
+      MapAnimationOptions(owner, duration, startDelay, interpolator, animatorListener)
   }
 
   /**
-   * Hash code method.
+   * Equals method.
    */
-  override fun hashCode(): Int {
-    var result = duration ?: 0
-    result = 31 * result + (owner?.hashCode() ?: 0)
-    result = 31 * result + (interpolator?.hashCode() ?: 0)
-    result = 31 * result + (animatorListener?.hashCode() ?: 0)
-    return result.toInt()
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (javaClass != other?.javaClass) return false
+
+    other as MapAnimationOptions
+
+    if (owner != other.owner) return false
+    if (duration != other.duration) return false
+    if (startDelay != other.startDelay) return false
+    if (interpolator != other.interpolator) return false
+    if (animatorListener != other.animatorListener) return false
+
+    return true
   }
 
   /**
-   * Equals method.
+   * Hash code method.
    */
-  override fun equals(other: Any?): Boolean {
-    if (this === other) {
-      return true
-    }
-    if (other == null || javaClass != other.javaClass) {
-      return false
-    }
-    val that = other as MapAnimationOptions
-    if (that.owner != owner) {
-      return false
-    }
-    if (that.duration != duration) {
-      return false
-    }
-    if (that.interpolator != interpolator) {
-      return false
-    }
-    if (that.animatorListener != animatorListener) {
-      return false
-    }
-    return true
+  override fun hashCode(): Int {
+    var result = owner?.hashCode() ?: 0
```

Comment:
just a question
if we will create two objects and set all fields to `null` we will get collision?

Reply:
In this situation these two objects are equal, right? Then they should have the same hashcode.

Reply:
I can create one `MapAnimationOptions` with all null fields except `duration` and set it to `0`
and other object `MapAnimationOptions` with all null fields except `startDelay` and set it to `0` 
this object not equals but hashCode will be the same. And as a user I want to put this objects inside hash* structure.
So maybe change `?: 0)` to some value like `42`?
wdyt?

Reply:
@alexander-kulikovskii I have a broader q - why are our hash-code equals tests passing then if we have concerns here?

Reply:
it's a very good question. probably `EqualsVerifier` not fully configured or even didn't allow catch such cases